File: backend/app/schemas/prompt.py
from pydantic import BaseModel, Field
from uuid import UUID
from datetime import datetime


class PromptCreate(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    description: str | None = None
    prompt_content: str = Field(min_length=1)
    category: str | None = None


class PromptUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=200)
    description: str | None = None
    prompt_content: str | None = None
    category: str | None = None


class PromptResponse(BaseModel):
    id: UUID
    title: str
    description: str | None
    prompt_content: str
    category: str | None
    is_system: bool
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/prompt.ts
export interface Prompt {
  id: string;
  title: string;
  description: string | null;
  prompt_content: string;
  category: string | null;
  is_system: boolean;
  created_at: string;
}

export interface Template {
  id: string;
  title: string;
  description: string | null;
  domain: string | null;
  framework: string | null;
  template_content: string;
  is_system: boolean;
  created_at: string;
}
